chromium_src/chrome/browser/sync/profile_sync_service_factory.cc

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/sync/driver/profile_sync_service.h"
+
+class Profile;
+
+namespace {
```

Comment:
seems better to put the `.cc` include after the method definition and avoid this forward decl?

Reply:
then we have to include whatever headers we need in the method definition

Reply:
I'm not sure what you mean, the method definition is in this same file. Whether you include them before or after https://github.com/brave/brave-core/pull/2010/files#diff-1aeebeb2e5f4cb67b74d12f233b9dac9R16 doesn't make any difference

Reply:
for this case I guess it shouldn't matter because all it does in the function is just construct. Some methods do more than that and the headers it needs are included in the cc file only, if you put it before .cc, you have to include those headers.

I will make the change

Reply:
addressed in b6f353557be0b4fc8d611c56c304025baf6356bd